src/internal/restgateway/restgateway.go

```diff
@@ -0,0 +1,114 @@
+package restgateway
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+
+	pfsgw "github.com/pachyderm/pachyderm/v2/src/pfs"
+	ppsgw "github.com/pachyderm/pachyderm/v2/src/pps"
+
+	admingw "github.com/pachyderm/pachyderm/v2/src/admin"
+	authgw "github.com/pachyderm/pachyderm/v2/src/auth"
+	debuggw "github.com/pachyderm/pachyderm/v2/src/debug"
+	enterprisegw "github.com/pachyderm/pachyderm/v2/src/enterprise"
+	identitygw "github.com/pachyderm/pachyderm/v2/src/identity"
+	licensegw "github.com/pachyderm/pachyderm/v2/src/license"
+	proxygw "github.com/pachyderm/pachyderm/v2/src/proxy"
+	transactiongw "github.com/pachyderm/pachyderm/v2/src/transaction"
+	versiongw "github.com/pachyderm/pachyderm/v2/src/version/versionpb"
+	workergw "github.com/pachyderm/pachyderm/v2/src/worker"
+)
+
+func NewRestGatewayMux(pachClientFactory func(context.Context) *client.APIClient) http.Handler {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	client := pachClientFactory(ctx)
+
+	mux := runtime.NewServeMux(runtime.WithIncomingHeaderMatcher(func(s string) (string, bool) {
+		if s != "Content-Length" {
+			return s, true
+		}
+		return s, false
+	}))
+
+	err := ppsgw.RegisterAPIHandler(ctx, mux, client.ClientConn())
+	if err != nil {
+		log.Error(ctx, "restgateway: failed to register pps api with grpc-gateway")
```

Comment:
```go
ctx := pctx.Child(ctx, "restgateway")
log.Info(ctx, "something")
```
is the idiomatic way to take on "restgateway: " to ever message.  Actually, it changes the "logger name", and it stacks, so this might be "startup(fullpachd).http.restgateway" or something.